packages/gestalt/src/NumberField.js

```diff
@@ -0,0 +1,152 @@
+// @flow strict
+import { forwardRef, type Node } from 'react';
+import InternalTextField from './InternalTextField.js';
+
+// <input> deals with strings, but we only want numbers for this component.
+// So we parse what we get from InternalTextField and we stringify what we give it.
```

Comment:
nit 
So we stringify what we give to InternalTexField and then parse what we get from InternalTextField